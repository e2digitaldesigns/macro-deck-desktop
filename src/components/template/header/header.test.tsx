import "@testing-library/jest-dom";
import { render, fireEvent } from "@testing-library/react";
import TemplateHeader from "./header";

describe("<Template Header Component/>", () => {
  it("Should render without errors", () => {
    const { getByTestId } = render(<TemplateHeader />);
    const component = getByTestId("template-header-component");
    expect(component).toBeTruthy();

    const fullScreenButton = getByTestId("template-header-full-screen");
    expect(fullScreenButton).toBeVisible();
  });

  it("Should match snapshot", () => {
    const { getByTestId } = render(<TemplateHeader />);
    const component = getByTestId("template-header-component");
    expect(component).toMatchSnapshot();
  });

  it("fullScreenExitButton should be visible", () => {
    const { getByTestId } = render(<TemplateHeader />);
    const fullScreenButton = getByTestId("template-header-full-screen");
    fireEvent.click(fullScreenButton);
    const fullScreenExitButton = getByTestId(
      "template-header-full-screen-exit"
    );
    expect(fullScreenExitButton).toBeVisible();
  });
});
