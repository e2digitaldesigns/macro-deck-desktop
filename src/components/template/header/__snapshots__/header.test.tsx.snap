// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Template Header Component/> Should match snapshot 1`] = `
<header
  class="sc-bdfBwQ dwSDNm"
  data-testid="template-header-component"
>
  <ul
    class="sc-gsTCUz flOpgs"
    data-testid="template-header-menu"
  >
    <li>
      File
    </li>
    <li>
      Edit
    </li>
    <li>
      Selection
    </li>
    <li>
      View
    </li>
    <li>
      Go
    </li>
  </ul>
  <ul
    class="sc-gsTCUz sc-dlfnbm flOpgs dsKvub"
    data-testid="template-header-options"
  >
    <li
      data-testid="template-header-minimize-button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="RemoveIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13H5v-2h14v2z"
        />
      </svg>
    </li>
    <li
      data-testid="template-header-full-screen-button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="FullscreenIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 14H5v5h5v-2H7v-3zm-2-4h2V7h3V5H5v5zm12 7h-3v2h5v-5h-2v3zM14 5v2h3v3h2V5h-5z"
        />
      </svg>
    </li>
    <li
      data-testid="template-header-full-close-button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CloseIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </li>
  </ul>
</header>
`;
